### PR TITLE
aarch64 implement brz,brnz,br_icmp for i128 values

### PR DESCRIPTION
**Pull request by afonso360**
Hey,

This PR implements a few branches for i128 values.

It also cleans up a bit of duplicated logic for lowering icmp

### CODE REVIEW

File: cranelift/filetests/filetests/isa/aarch64/condbr.clif [hunk 1]
```diff
@@ -253,3 +253,357 @@ block1:
...
+
+block1:
+    nop
+    return
+}
+
+; check: stp fp, lr, [sp, #-16]!
+; nextln: mov fp, sp
+; nextln: orr x0, x0, x1
+; nextln: cbz x0, label1 ; b label2
```
Comment (cfallin) [inline cranelift/filetests/filetests/isa/aarch64/condbr.clif:271]: I wonder: would it be better here to do an `ORRS` into `XZR` and then branch on the `Z` flag (here and for brnz below)? That would avoid the need to clobber a register, reducing pressure.

**Review by cfallin — APPROVED** [review]
Thanks for this, especially the very thorough tests!

I have one comment below on a possible optimization and am curious what you think, but am happy with this either way.

File: cranelift/filetests/filetests/isa/aarch64/condbr.clif (hunk 1 above)
Comment (afonso360) [reply cranelift/filetests/filetests/isa/aarch64/condbr.clif:271]: I think `ORRS` is no longer an instruction in aarch64, at least I'm not finding it in the [instruction listing](https://developer.arm.com/documentation/dui0802/a/A64-General-Instructions/A64-general-instructions-in-alphabetical-order?lang=en)

Comment (cfallin) [reply cranelift/filetests/filetests/isa/aarch64/condbr.clif:271]: Ah! I have been deceived by old ARM32 online docs, then; in this case, I agree this is probably the best we can do. Thanks!

**Pull request merged**